File: web/src/types/enums/Sockets.ts
import { SocketResponse } from "../interfaces/SocketResponse";
import { User } from "../interfaces/User";
import { AsientosRes } from "../res/AsientosRes";

export enum Sockets {
  CONNECT = "connect",
  COUNT = "count",
  DISCONNECT = "disconnect",
  MESSAGE = "message",
  USERMODIFY = "userModify",
  USERGROUP = "userGroup",

  ASIENTOLOCAL = "asientoLocal",
  ASIENTOPOST = "asientoPost",
  ASIENTOPUT = "asientoPut",
  ASIENTODELETE = "asientoDelete",
}

export type SocketCallbacks = {
  [Sockets.CONNECT]: (res: SocketResponse<null>) => void;
  [Sockets.COUNT]: (res: SocketResponse<null>) => void;
  [Sockets.DISCONNECT]: (res: SocketResponse<null>) => void;
  [Sockets.MESSAGE]: (res: SocketResponse<null>) => void;
  [Sockets.USERMODIFY]: (res: SocketResponse<User>) => void;
  [Sockets.USERGROUP]: (res: SocketResponse<null>) => void;

  [Sockets.ASIENTOLOCAL]: (res: SocketResponse<null>) => void;
  [Sockets.ASIENTOPOST]: (res: SocketResponse<AsientosRes>) => void;
  [Sockets.ASIENTOPUT]: (res: SocketResponse<AsientosRes>) => void;
  [Sockets.ASIENTODELETE]: (res: SocketResponse<number>) => void;
};

File: server/Constants/Sockets.cs
namespace server.Constants
{
    public static class Sockets
    {
        public static class Types
        {
            public static string CONNECT { get; } = "connect";
            public static string COUNT { get; } = "count";
            public static string DISCONNECT { get; } = "disconnect";
            public static string MESSAGE { get; } = "message";
            public static string USERMODIFY { get; } = "userModify";
            public static string USERGROUP { get; } = "userGroup";
        }

        public static class Asiento
        {
            public static string LOCAL { get; } = "asientoLocal";
            public static string POST { get; } = "asientoPost";
            public static string PUT { get; } = "asientoPut";
            public static string DELETE { get; } = "asientoDelete";
        }
    }
}
